README.md

```diff
@@ -30,153 +31,238 @@ addOne(2); // 3
+
+The `?` placeholder token can be supplied one or more times at the top level of the argument list of
+a _CallExpression_, _CallMemberExpression_, or `new` (e.g. `f~(?)` or `o.f~(?)`). `?` is **not**
+an expression, rather it is a syntactic element that indicates special behavior (much like how
+`` `...` AssignmentExpression `` indicates spread, yet is itself not an expression).
 
 ```js
 // valid
-f(x, ?)           // partial application from left
-f(?, x)           // partial application from right
-f(?, x, ?)        // partial application for any arg
-o.f(x, ?)         // partial application from left
-o.f(?, x)         // partial application from right
-o.f(?, x, ?)      // partial application for any arg
-super.f(?)        // partial application allowed for call on |SuperProperty|
+f~(x, ?)          // partial application from left
+f~(?, x)          // partial application from right
+f~(?, x, ?)       // partial application for any arg
+o.f~(x, ?)        // partial application from left
+o.f~(?, x)        // partial application from right
+o.f~(?, x, ?)     // partial application for any arg
+super.f~(?)       // partial application allowed for call on |SuperProperty|
+new C~(?)         // partial application of constructor
 
 // invalid
-f(x + ?)          // `?` not in top-level Arguments of call
+f~(x + ?)         // `?` not in top-level Arguments of call
 x + ?             // `?` not in top-level Arguments of call
-?.f()             // `?` not in top-level Arguments of call
-new f(?)          // `?` not supported in `new`
-super(?)          // `?` not supported in |SuperCall|
+?.f~()            // `?` not in top-level Arguments of call
+super~(?)         // `?` not supported in |SuperCall|
 ```
 
-# Semantics
+In addition the `?` placeholder token can be followed by a decimal integer value &ge; 0 indicating a fixed
+ordinal position (i.e., `?0`). Ordinal placeholders can be repeated multiple times within a partial application,
+but still point to the same unbound argument. If a partial application contains a mix of non-ordinal placeholders 
+and ordinal placeholders, the non-ordinal placeholders are implicitly numbered to fill in any gaps:
 
-The `?` placeholder token can only be used in an argument list of a call expression, or as the only
-token in a placeholder of a template expression or tagged template expression. When present, the 
-result of the call is a new function with a parameter for each `?` token in the argument list. 
-Any non-placeholder expression in the argument list becomes fixed in its position. This is 
-illustrated by the following syntactic conversion:
+```js
+const printArgs = (a = "a", b = "b", c = "c") => console.log(`${a}, ${b}, ${c}`);
+printArgs();                            // prints: a, b, c
+printArgs(1, 2, 3);                     // prints: 1, 2, 3
+
+const swapAC = printArgs~(?2, ?, ?0);
+swapAC(1, 2, 3);                        // prints: 3, 2, 1
+
+const ignoreBC = printArgs~(?2);
```

Comment:
This is also confusing. This desugars into something like the following, right?:

```js
// need 3 params, because `?2` was specified
const ignoreBC = (_0, _1, _2) => {
  // only _2 was partialed
  return printArgs(_2);
};
```